rpg.py

```diff
@@ -41,56 +54,136 @@ def deal_damage(self, other_player):
         return other_player
 
     def get_all_enemies(self, players: List) -> List[int]:
+        """returns the location of all enemy players in a list"""
         return [
             index for index, player in enumerate(players)
             if not self.is_ally(player) and player.is_alive()
         ]
 
+    def get_all_allies(self, players: List) -> List[int]:
+        """returns the locations of all ally players in a list"""
+        return[
+            index for index, player in enumerate(players)
+            if self.is_ally(player1) and self.is_alive()
+        ]
+
     def act(self, players: List) -> List:
+        """how one player acts upon another, eg. an attack"""
         all_enemy_locations = self.get_all_enemies(players)
         if all_enemy_locations:
             if self.is_alive:
-                print(f'{self.name}\'s turn!')
-            targeted_index = all_enemy_locations[0]
-            targeted_player = players[targeted_index]
-            damaged_player = self.deal_damage(targeted_player)
-            players[targeted_index] = damaged_player
+                targeted_index = all_enemy_locations[0]
```

Comment:
You should check if `self.is_alive` before looking at the list, to save computation time of iterating through the list.

Also, you never actually called `self.is_alive`, it should be:
```
if self.is_alive():
```